master/internal/job/job.go

```diff
@@ -2,111 +2,103 @@ package job
 
 import (
 	"fmt"
-	"slices"
 	"sort"
 	"strings"
 	"sync"
 
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
 
 	"github.com/determined-ai/determined/master/internal/rm"
 	"github.com/determined-ai/determined/master/internal/rm/rmerrors"
 	"github.com/determined-ai/determined/master/internal/sproto"
-	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/master/pkg/model"
 	"github.com/determined-ai/determined/proto/pkg/jobv1"
 )
 
-// Manager manages jobs.
-type Manager struct {
-	mu        sync.Mutex
-	rm        rm.ResourceManager
-	actorByID map[model.JobID]*actor.Ref
-	system    *actor.System
-	syslog    *logrus.Entry
+// GenericJob is the interface for commands/experiments types that implement
+// the job service.
+type GenericJob interface {
```

Comment:
nit: I feel like we over-use the word "generic" in our code base when there is not really a specialized version. personally I would prefer to name it Job or JobInterface, and genericByID as something like jobByID

Reply:
I'm pro `Job`! Would love to hear what @stoksc thinks too